### PR TITLE
I63 call exact and fuzzy search separately

### PR DESCRIPTION
**Pull request by hugeliveux**
I left the API intact. Adding more types of search in the future (#62) only requires adding more `specifiedSearch`

Clear search result before each search, because when typing a second search when the first one is already there, the header text will show a wrong message (e.g. “共找到0个精确结果 (关键字: second search)” while showing a bunch of results to the first search)

Added a center style in the spinner component. I think this could be set default.

Please help me with my first push, all kinds of error are very likely to show up. Ty :)

### CODE REVIEW

**Comment by hgbe33** [comment]
Sorry, I rebased your branch onto the latest master and made some API changes based on your needs. Hope that helps.

**Comment by hugeliveux** [comment]
I found the new changes. How did that "rebase" work exactly? I first merged the newest master into my local i63 branch, then told I was "behind origin", then I merge origin/i63 into my local (don't know why that is called a "merge"), then solved the conflicts, and now I'm ok. Still a bit confused...

The build works locally. I hope everything is ok.

File: pages/search/index.js [hunk 1]
```diff
@@ -405,6 +387,60 @@ class Search extends React.Component {
...
+
+    fetch(searchURL)
+      .then((res) => res.json())
+      .then((result) => {
+        localStorage.setItem(Refs.region, result.region)
+        this.updateResults()[mode](result)
+      }, this.errorHandler)
+  }
+
+  updateResults() {
```
Comment (kkhta) [inline pages/search/index.js:422]: I presume you wrote this as a function to make sure that it's bound to `<Search/>` right? We can actually just declare `updateResults` as a plain object here. Notice that when it's used, you have `this.updateResults[]()`, where `this` resides in `specifiedSearch`, thus is already bound to `<Search/>`.

Comment (hugeliveux) [reply pages/search/index.js:422]: Ah~~ such enlightening

File: pages/search/index.js [hunk 2]
```diff
@@ -337,58 +337,40 @@ class Search extends React.Component {
...
-      encodeURIComponent(keyword) +
-      "&sortBy=" +
-      encodeURIComponent(this.state.sortBy) +
-      "&type=" +
-      encodeURIComponent(this.state.type) +
-      "&region=" +
-      encodeURIComponent(this.state.region) +
-      "&year=" +
-      encodeURIComponent(this.state.year)
+    const hasKeyword = this.state.keyword !== ""
```
[the remark this answers is not in the document]
Comment (hugeliveux) [reply pages/search/index.js:354]: Cool trick!
However, check out this article: https://roderick.dk/2018/09/10/avoid-double-exlamations/
Using `Boolean` to force type is also neat. I guess I'll stick with current style for now.

**Pull request merged**